asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py

```diff
@@ -242,13 +242,67 @@ def get_interaction_color(intn_type) -> str:
             return "#8C4099"
         else:
             raise ValueError(f"Interaction type {intn_type} not recognized.")
+        
+    def is_backbone_residue(self, x, y, z) -> bool:
```

Comment:
this function is gnarly, if you can think of a more elegant way of doing this we should implement it. In any case, currently this is very fast (practically instant) and is catching all interactions.